samples/arena-samples/mpi/mpi_run.py

```diff
@@ -9,52 +9,40 @@
   name='pipeline to run mpi job',
   description='shows how to run mpi job.'
 )
-def sample_mpirun_pipeline(image="registry.cn-hangzhou.aliyuncs.com/tensorflow-samples/horovod:0.13.11-tf1.10.0-torch0.4.0-py3.5",
+def mpirun_pipeline(image="uber/horovod:0.13.11-tf1.10.0-torch0.4.0-py3.5",
 						   batch_size="64",
-						   optimizer='momentum'):
+						   optimizer='momentum',
+               sync_source='https://github.com/tensorflow/benchmarks.git',
+               git_sync_branch='cnn_tf_v1.9_compatible',
+               data='user-susan:/training',
+               gpus=1,
+               workers=1,
+               cpu_limit='2',
+               metric='images/sec',
+               memory_limit='10Gi'):
   """A pipeline for end to end machine learning workflow."""
-  data=["user-susan:/training"]
-  env=["NCCL_DEBUG=INFO", ""]
-  gpus=1
+
+  env = ['NCCL_DEBUG=INFO','GIT_SYNC_BRANCH={0}'.format(git_sync_branch)]
 
   train=arena.mpi_job_op(
   	name="all-reduce",
   	image=image,
   	env=env,
+    data=[data],
+    workers=workers,
+    sync_source=sync_source,
+    gpus=gpus,
+    cpu_limit=cpu_limit,
+    memory_limit=memory_limit,
+    metrics=[metric],
   	command="""
-  	mpirun python /benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --model resnet101 \
+  	mpirun python code/benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --model resnet101 \
   	--batch_size {0}  --variable_update horovod --optimizer {1}\
   	--summary_verbosity=3 --save_summaries_steps=10
   	""".format(batch_size, optimizer)
   )
 
   
 if __name__ == '__main__':
-  parser = argparse.ArgumentParser()
-  parser.add_argument('--image', type=str,
-                      default="registry.cn-hangzhou.aliyuncs.com/tensorflow-samples/horovod:0.13.11-tf1.10.0-torch0.4.0-py3.5",
-                      help='docker image.')
-  parser.add_argument('--batch_size', type=str, default="64",
-                      help='batch size.')
-  parser.add_argument('--optimizer', type=str, default="momentum",
-                      help='optimizer.')
-  FLAGS, unparsed = parser.parse_known_args()
-
-  image = FLAGS.image
-  batch_size = FLAGS.batch_size
-  optimizer = FLAGS.optimizer
-
-  EXPERIMENT_NAME="resnet101-allreduce"
-  RUN_ID="run"
-  KFP_SERVICE="ml-pipeline.kubeflow.svc.cluster.local:8888"
   import kfp.compiler as compiler
   compiler.Compiler().compile(sample_mpirun_pipeline, __file__ + '.tar.gz')
```

Comment:
Probably forgot to change this to `mpirun_pipeline`?

Reply:
Thank you for spotting it. In this sample, I'm using `dsl-compile`.